collecting-parameter/README.md

```diff
@@ -0,0 +1,165 @@
+
+```java
+public class App {
+
+  public static void main(String[] args) {
+    List<Apple> apples = new ArrayList<>();
+
+    apples.add(new Apple(10, "a"));
+    apples.add(new Apple(20, "b"));
+    apples.add(new Apple(15, "c"));
+
+    StringBuilder result = new StringBuilder();
+    for(Apple iter : apples) {
+      result.append("apple weight: ");
+      result.append(iter.weight);
+      result.append(" \n");
+    }
+  }
+}
+```
+
+Some times the way we want to accumulate the information in the object might be very complicated, or we want to collect information from different kinds of objects .We can use collecting parameter pattern to simplify it.
+
+First modify the `Apple.java`:
+```java
+public class Apple {
+  int weight;
+  String type;
+
+  public Apple(int weight, String type) {
+    this.weight = weight;
+    this.type = type;
+  }
+
+  public void writeWeightTo(List<String> weightInfo) {
+    String appleWeight = "apple weight: " + weight + " \n";
+    weightInfo.add(appleWeight);
+  }
+
+  public void writeTypeTo(List<String> typeInfo) {
+    String appleType = "apple type: " + type + " \n";
+    typeInfo.add(appleType);
+  }
+}
+```
+
+And now we can accumulate the information in targets easier by passing an object to their collecting methods.
+
+```java
+public class App {
+  /**
+   * Program entry point.
+   *
+   * @param args command line args
+   */
+  public static void main(String[] args) {
+    List<Apple> apples = new ArrayList<>();
+
+    apples.add(new Apple(10, "a"));
+    apples.add(new Apple(20, "b"));
+    apples.add(new Apple(15, "c"));
+
+    collectType(apples);
+    collectWeight(apples);
+  }
+
+  /**
+   * collect the Apple weight information by throwing weightInfo to apples.
+   *
+   * @param apples the apple list need to collect information from
+   */
+  public static void collectWeight(List<Apple> apples) {
+    List<String> weightInfo = new ArrayList<>();
+    for (Apple apple : apples) {
+      apple.writeWeightTo(weightInfo);
+    }
+    for (String message : weightInfo) {
+      LOGGER.info(message);
+    }
+  }
+
+  /**
+   * collect the Apple type information by throwing typeInfo to apples.
+   *
+   * @param apples the apple list need to collect information from
+   */
+  public static void collectType(List<Apple> apples) {
+    List<String> typeInfo = new ArrayList<>();
+    for (Apple apple : apples) {
+      apple.writeTypeTo(typeInfo);
+    }
+    for (String message : typeInfo) {
+      LOGGER.info(message);
+    }
+  }
+}
+
+
+```
+
+## Class diagram
+![alt text](./etc/collectingparameter.urm.png "Adapter class diagram")
+
+## Applicability
+Use the collecting parameter pattern when
+
+* You have a single bulky method that accumulates information to a local variable
+## Consequences:
+Collecting parameter pattern can:
+
+*	Helps transform bulky methods into smaller, simpler methods.
+*   make resulting code run faster.
```

Comment:
Correct the formatting and provide more references on how can it make the code run faster would really help us understand.

```suggestion
* Helps transform bulky methods into smaller, simpler methods.
* Make the resulting code run faster.
```